lxd/endpoints/metrics.go

```diff
@@ -0,0 +1,86 @@
+package endpoints
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/lxc/lxd/lxd/util"
+	"github.com/lxc/lxd/shared"
+	"github.com/lxc/lxd/shared/logger"
+)
+
+func metricsCreateListener(address string, cert *shared.CertInfo) (net.Listener, error) {
```

Comment:
@monstermunchkin please can you add the fix from https://github.com/lxc/lxd/pull/9259 to this new function too.